text/083-boost-stdlib-dev/text.md

```diff
@@ -0,0 +1,736 @@
+
+- Identify consistent stdlib components that can be used independently
+  from each other and that would be worth distributing as separate
+  packages. Identify their maintainers and give them freedom to define
+  the future of the components they maintain, in the limits set by the
+  Coq Platform charter. Allow maintainers to extract stdlib components
+  to maintain and evolve them outside the core Coq repository and to have
+  their own release schedule and versioning scheme, in case they wish to
+  do so.
+
+- Extract the prelude + a minimum set of components that alternative
+  general libraries like MathComp and coq-stdpp need as a basis.
+  Make sure that this reduced core stdlib component can evolve to allow
+  other libraries to use newer features of Coq (like universe polymorphism,
+  SProp and primitive projections).
+  Stop including any other stdlib components as part of the `coq` opam
+  package and encourage maintainers of Coq packages in other distributions
+  to do the same.
+
+# Detailed design
+
+## Mono vs Dual Repository
+
+- In the current situation (let's call it *monorepo*), Coq and the
+  stdlib live int the same repository (i.e. https://github.com/coq/coq ).
+  Pros:
+  - Coq and the stdlib are naturally kept in sync, without the
+    need for any overlay mechanism (although developments of other
+    libraries proves that this is more a matter of comfort than a vital
+    requirement).
+- The current CEP offers to split most of the current stdlib into another
+  repo (let's call it *dualrepo*). Pros:
+  - Coq and stdlib development teams are mostly disjoint, the two repos
+    would reflect that and enable both teams to focus on their main point 
+    of interest (the small intersection could of course remain active in
+    both teams).
+  - The current Coq repo is scary for potential new stdlib contributors:
+    it contains lot of OCaml code, enforces heavy development processes
+    and the few stdlib PRs that get opened anyway often don't get timely
+    reviews because most Coq developers don't care about them. We thus
+    likely loose most potential new contributors to the stdlib and most
+    Coq users end up with the feeling that the stdlib is not something
+    that can be practically improved, do their own developments on their
+    side, and never contribute anything back. Having its own repo would
+    make the stdlib more welcoming for newcomers and hopefully help
+    get it out of this mostly-stale situation it is suffering for years.
+  - Separate repos enable each team to roll out their own release schedule
+    independently. Stdlib developers no longer need to wait for the next
+    Coq release, neither are they forced to release every time Coq does
+    (provided the last stdlib release remains compatible with the new Coq release).
+  - Minor point: a separate repo wouldn't force Coq developers to compile
+    the stdlib everytime, for instance just to run the test-suite.
```

Comment:
> I think Coq developers almost always need to compile stdlib, particularly to run the test suite.

In fact only a small portion of the test-suite tests the stdlib, most of it is testing Coq itself. That point is adressed a few paragraphs below.

>  OCaml changes may break the current stdlib or other libraries--and we want to know that sooner rather than later.

As you point, nothing specific to the stdlib here. It's the same as any other lib in the CI.

Reply:
Successful compilation of stdlib and other libraries is an essential test of Coq, right?

Reply:
Yes, that's what the CI is for, this proposal will of course not change anything with that respect.

Reply:
So IMO the "minor point" isn't helpful or persuasive.

Reply:
It indeed seems some developers don't really care. I personally do (a bit). For instance, when I do change something in printing, I want to rerun output tests without having to recompile the whole stdlib (that doesn't test anything regarding outputs).

Reply:
@proux01 we could achieve this by merging Ali's PR to port the test-suite to dune. In fact, that PR had many other benefits (and closed a lot of bugs)